letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -451,6 +451,34 @@ OldVenvExists() {
     [ -n "$OLD_VENV_PATH" -a -f "$OLD_VENV_PATH/bin/letsencrypt" ]
 }
 
+# Given python path, version 1 and version 2, check if version 1 is outdated compared to version 2.
+# Unofficial versions (eg. 0.28.0.dev0) will be treated specifically by printing "UNOFFICIAL".
+# Otherwise, print "OUTDATED" if version 1 is outdated, or "UP_TO_DATE" if not.
+# This function relies only on installed python environment (2.x or 3.x) by certbot-auto.
+CompareVersions() {
+    "$1" - "$2" "$3" << "UNLIKELY_EOF"
```

Comment:
This is a really clever way of delegating to another language without having to make a file!

Reply:
Thanks a lot! I was tempted to do the same for the others temporary files that are python scripts, including your pipstrap, before remembering that I should not overcharge a PR with unrelated work.

But if you think it is quite elegant, I will be happy to do it in another PR!